compiler/src/org.graalvm.compiler.truffle.compiler/src/org/graalvm/compiler/truffle/compiler/phases/inlining/DefaultInliningPolicy.java

```diff
@@ -132,12 +132,16 @@ private void inline(CallTree tree) {
         }
     }
 
-    private static boolean isExcludedMethod(CallNode candidate, String[] excludedMethods) {
+    static boolean isExcludedMethod(CallNode candidate, String[] excludedMethods) {
+        // Removes split information from name
+        //   e.g. "Object#foo <split-1234>" => "Object#foo"
+        final String candidateWithoutSplitInfo = candidate.getTruffleAST().getName()
+                .replaceAll(" \\<[^()]*\\>", "");
```

Comment:
This regex seems a bit too aggressive no? It should be only if ` <split-1234>` is on the end of the string. I'd recommend something like ` <split-[0-9|a-f]*>$`